kani-driver/src/concrete_playback.rs

```diff
@@ -11,13 +11,44 @@ use anyhow::{Context, Result};
 use concrete_vals_extractor::{extract_harness_values, ConcreteVal};
 use kani_metadata::HarnessMetadata;
 use std::collections::hash_map::DefaultHasher;
+use std::env;
 use std::ffi::OsString;
-use std::fs::{self, File};
+use std::fs::{self, remove_file, File};
 use std::hash::{Hash, Hasher};
-use std::io::{Read, Write};
-use std::path::Path;
+use std::io::{BufRead, BufReader, BufWriter, Error, Write};
+use std::path::{Path, PathBuf};
 use std::process::Command;
 
+struct TempFile {
+    file: File,
+    temp_path: PathBuf,
+}
+
+impl TempFile {
+    fn new() -> Result<Self, Error> {
```

Comment:
nit: rename this to `try_new`. Usually `try_new()` means that the creation can fail.